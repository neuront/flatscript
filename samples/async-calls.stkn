func asyncTextGen(input, cb)
    if true
        setTimeout(%, 1)
        if input.length < 4
            cb(input + ' is quite short')
        else
            cb(input + ' is quite long')
    return input.length

console.log(asyncTextGen('cat', %text))
console.log(text)
console.log(asyncTextGen(text, %moreText))
console.log(moreText)

func cancelAtFirstTime(timeoutHandle)
    clearTimeout(timeoutHandle)

cancelAtFirstTime(setInterval(%, 1))

console.log('End')
