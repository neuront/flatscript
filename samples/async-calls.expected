3
cat is quite short
18
cat is quite short is quite long
End
